app/src/main/java/de/danoeh/antennapod/fragment/gpodnet/GpodnetMainFragment.java

```diff
@@ -26,63 +35,59 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         toolbar.setTitle(R.string.gpodnet_main_label);
         ((AppCompatActivity) getActivity()).setSupportActionBar(toolbar);
 
-        ViewPager viewPager = root.findViewById(R.id.viewpager);
-        GpodnetPagerAdapter pagerAdapter = new GpodnetPagerAdapter(getChildFragmentManager());
+        ViewPager2 viewPager = root.findViewById(R.id.viewpager);
+        GpodnetPagerAdapter pagerAdapter = new GpodnetPagerAdapter(this);
         viewPager.setAdapter(pagerAdapter);
 
         // Give the TabLayout the ViewPager
         TabLayout tabLayout = root.findViewById(R.id.sliding_tabs);
-        tabLayout.setupWithViewPager(viewPager);
+
+        new TabLayoutMediator(tabLayout, viewPager, (tab, position) -> {
+            switch (position) {
+                case POS_TAGS:
+                    tab.setText(R.string.gpodnet_taglist_header);
+                    break;
+                case POS_TOPLIST:
+                    tab.setText(R.string.gpodnet_toplist_header);
+                    break;
+                default:
+                case POS_SUGGESTIONS:
+                    tab.setText(R.string.gpodnet_suggestions_header);
+                    break;
+            }
+        }).attach();
 
         return root;
     }
 
-    public class GpodnetPagerAdapter extends FragmentPagerAdapter {
-        private static final int NUM_PAGES = 2;
-        private static final int POS_TOPLIST = 0;
-        private static final int POS_TAGS = 1;
-        private static final int POS_SUGGESTIONS = 2;
```

Comment:
Not in scope of this change but weird that NUM_PAGES is 2 but we have three pages here. Guess better to fix it later.

Reply:
@ByteHamster while this may take time please have a look if NUM_PAGES = 2 is really desired and nothing has got hide by mistake if possible thanks.

Reply:
I created issue #4093 for this.